c10/core/TensorImpl.h

```diff
@@ -477,7 +477,12 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
       Storage&& storage,
       DispatchKeySet,
       const caffe2::TypeMeta data_type);
-
+  // Used to construct a nested_tensor view
+  TensorImpl(
```

Comment:
cc @ezyang about adding new constructor here?

Reply:
It's probably the wrong thing to do. I'll take a read